cfme/tests/cloud/test_providers.py

```diff
@@ -1397,3 +1398,34 @@ def test_add_ec2_provider_with_sts_assume_role(appliance, ec2_provider_with_sts_
     """
     ec2_provider_with_sts_creds.create()
     ec2_provider_with_sts_creds.validate()
+
+
+@test_requirements.ec2
+@pytest.mark.meta(automates=[1658207])
+@pytest.mark.provider([EC2Provider], scope="function", selector=ONE)
+def test_add_second_provider(setup_provider, provider, request):
+    """
+        Bugzilla: 1658207
+
+        Polarion:
+        assignee: mmojzis
+        casecomponent: Cloud
+        initialEstimate: 1/3h
+        caseimportance: high
+        casecomponent: Cloud
+        testSteps:
+            1. Go to Compute -> Cloud -> Providers
+            2. Add EC2 Provider
+            3. Add another EC2 Provider
+        expectedResults:
+            1.
+            2. Provider should be successfully added.
+            3. Provider should be successfully added.
+    """
+    second_provider = provider
```

Comment:
This cannot be done via simple assignment, as it modifies the original instance.

```
In [2]: provider                                                                                                              
Out[2]: EC2Provider(name='ec2-west', key='ec2west', zone='default', region='us-west-1', region_name='US West (N. California)')

In [3]: second_provider=provider                                                                                              

In [4]: second_provider.name='junk name'                                                                                      

In [5]: provider                                                                                                              
Out[5]: EC2Provider(name='junk name', key='ec2west', zone='default', region='us-west-1', region_name='US West (N. California)')

```

It has to be done using `deepcopy`

Reply:
For those following along at home - deepcopy actually can't be used with the provider instances, but the `cfme.utils.providers.get_crud` method can be used to fetch another provider instance using the key.

Reply:
Reworked with using get_crud.